punchcut/src/source/glyf/scaler.rs

```diff
@@ -0,0 +1,640 @@
+use super::math;
+use super::{Context, Outline, Point};
+use crate::{Error, Hinting, NormalizedCoord, Result};
+
+use read_fonts::tables::{
+    glyf::{Anchor, CompositeGlyph, CompositeGlyphFlags, Glyf, Glyph, SimpleGlyph},
+    hmtx::Hmtx,
+    hvar::Hvar,
+    loca::Loca,
+};
+use read_fonts::types::{BigEndian, F2Dot14, GlyphId, Tag};
+use read_fonts::TableProvider;
+
+/// Limit for recursion when loading composite glyphs.
+const RECURSION_LIMIT: usize = 32;
+
+/// TrueType glyph scaler for a specific font and configuration.
+pub struct Scaler<'a> {
+    /// Backing context.
+    context: &'a mut Context,
+    /// Current font data.
+    font: Font<'a>,
+    /// Font identifier for the hinting cache.
```

Comment:
help the reader understand how such ids are created, preferably by having a type, at the very least a newtype not u64, and some lovely comments on said type

Reply:
Happy to newtype, but it needs to be flexible. We don't want to be responsible for generation or management of unique font ids. I mentioned above that I tried that with swash and it ended up being a huge source of friction for just about everybody trying to work with it (including me :)